internal/pkg/agent/application/paths/common.go

```diff
@@ -31,6 +31,8 @@ const (
 var ExternalInputsPattern = filepath.Join("inputs.d", "*.yml")
 
 var (
+	mu = sync.Mutex{}
```

Comment:
`mu sync.Mutex` should be fine.
i'd prefer pathLock to be more descriptive of what we're using it for